src/Entities/Project.php

```diff
@@ -20,25 +21,31 @@ public function __construct(int $id, string $name, int $client_id, ?DateTime $de
         $this->isOverdue = $this->calculateIsOverdue();
     }
 
-    private function calculateIsOverdue(): bool|null
+    private function calculateIsOverdue(): ?bool
     {
         $currentDate = new DateTime();
-        if($this->deadline !== null) {
+        if ($this->deadline !== null) {
             return $this->deadline < $currentDate;
         } else {
             return null;
         }
     }
 
     // getIsOverdue() is purely for unit testing
-    public function getIsOverdue(): bool|null
+    public function getIsOverdue(): ?bool
     {
         return $this->isOverdue;
     }
 
     public function toAssociativeArrayFewerProperties(): array
     {
-        return ['id' => $this->id, 'name' => $this->name, 'client_id' => $this->client_id, 'deadline' => $this->deadline, 'overdue' => $this->isOverdue];
+
+        return [
+            'id' => strval($this->id),
+            'name' => $this->name, 
+            'client_id' => strval($this->client_id),
+            'deadline' => is_null($this->deadline) ? null : $this->deadline->format('d/m/Y'), 
```

Comment:
also does it make sense to run this logic in the constructor instead of down here?